mtags-shared/src/main/scala/scala/meta/internal/pc/InlayHints.scala

```diff
@@ -96,4 +95,42 @@ object InlayHints {
     buffer += LabelPart(tpeStr.substring(current, tpeStr.length))
     buffer.toList.filter(_.name.nonEmpty)
   }
+
+  def toData(uri: String, data: List[Either[String, l.Position]]): JsonElement =
+    gson.toJsonTree(
+      InlineHintData(
+        uri,
+        data.map {
+          case Left(str) => LabelPartData("string", str, null)
+          case Right(pos) => LabelPartData("position", null, pos)
+        }.toArray
+      )
+    )
+
+  def fromData(
+      json: JsonElement
+  ): (String, List[Either[String, l.Position]]) = {
+    val data = gson.fromJson(json, classOf[InlineHintData])
```

Comment:
We could use JsonParser and `.as[InlineHintData]` as a small improvement.

Reply:
It's in `mtags-shared` and `JsonParser` in `metals`. Even if I move `fromData` to `InlayHintResolveProvider`, I still need that `gson` value for writing (`toData`), so I'm not sure it's worth it.

Reply:
Ach, right. I didn't notice.